File: sysmon-android/app/src/main/java/com/sysmon/monitor/data/model/SystemMetrics.kt
package com.sysmon.monitor.data.model

import com.google.gson.annotations.SerializedName

/**
 * 与 sysmon-ws Rust 服务端 SystemMetrics 结构对应
 */
data class SystemMetrics(
    @SerializedName("timestamp")
    val timestamp: String = "",

    @SerializedName("cpu_usage_percent")
    val cpuUsagePercent: Float = 0f,

    @SerializedName("cpu_per_core")
    val cpuPerCore: List<Float> = emptyList(),

    @SerializedName("memory_usage_percent")
    val memoryUsagePercent: Float = 0f,

    @SerializedName("memory_used_mb")
    val memoryUsedMb: Long = 0L,

    @SerializedName("memory_total_mb")
    val memoryTotalMb: Long = 0L,

    @SerializedName("net_rx_kbps")
    val netRxKbps: Double = 0.0,

    @SerializedName("net_tx_kbps")
    val netTxKbps: Double = 0.0,

    // GPU 占用率（0~100），服务端不支持时字段不存在，此处默认 null 不报错
    @SerializedName("gpu_usage_percent")
    val gpuUsagePercent: Float? = null,
)

File: sysmon-ios/SysMon/Sources/Model/SystemMetrics.swift
import Foundation

// ══════════════════════════════════════════════════════════════════════════════
// SystemMetrics  —  对应 sysmon-ws Rust 服务端的 SystemMetrics 结构
// ══════════════════════════════════════════════════════════════════════════════

struct SystemMetrics: Codable {
    let timestamp:           String
    let cpuUsagePercent:     Float
    let cpuPerCore:          [Float]
    let memoryUsagePercent:  Float
    let memoryUsedMb:        Int64
    let memoryTotalMb:       Int64
    let netRxKbps:           Double
    let netTxKbps:           Double

    enum CodingKeys: String, CodingKey {
        case timestamp
        case cpuUsagePercent   = "cpu_usage_percent"
        case cpuPerCore        = "cpu_per_core"
        case memoryUsagePercent = "memory_usage_percent"
        case memoryUsedMb      = "memory_used_mb"
        case memoryTotalMb     = "memory_total_mb"
        case netRxKbps         = "net_rx_kbps"
        case netTxKbps         = "net_tx_kbps"
    }

    init(
        timestamp:          String = "",
        cpuUsagePercent:    Float  = 0,
        cpuPerCore:         [Float] = [],
        memoryUsagePercent: Float  = 0,
        memoryUsedMb:       Int64  = 0,
        memoryTotalMb:      Int64  = 0,
        netRxKbps:          Double = 0,
        netTxKbps:          Double = 0
    ) {
        self.timestamp          = timestamp
        self.cpuUsagePercent    = cpuUsagePercent
        self.cpuPerCore         = cpuPerCore
        self.memoryUsagePercent = memoryUsagePercent
        self.memoryUsedMb       = memoryUsedMb
        self.memoryTotalMb      = memoryTotalMb
        self.netRxKbps          = netRxKbps
        self.netTxKbps          = netTxKbps
    }
}
